crates/rust-analyzer/src/reload.rs

```diff
@@ -139,9 +139,10 @@ impl GlobalState {
                 "Proc-macros and/or build scripts have changed and need to be rebuilt.\n\n",
             );
         }
-        if self.fetch_build_data_error().is_err() {
-            status.health |= lsp_ext::Health::Warning;
+        if let Err(build_data_err) = self.fetch_build_data_error() {
+            status.health |= lsp_ext::Health::Error;
```

Comment:
This should stay a warning, even if build scripts fail we can still load the project.